config/puma/production.rb

```diff
@@ -17,3 +17,6 @@
   dynflow = ::Rails.application.dynflow
   dynflow.initialize! unless dynflow.config.lazy_initialization
 end
+
+# === Puma control rack application ===
+activate_control_app 'tcp://127.0.0.1:9293', { auth_token: '12345' }
```

Comment:
I chose port 9293 as per the [puma config recommendation](https://github.com/puma/puma#controlstatus-server). Do we have any restrictions on choosing the port numbers in foreman?

Reply:
Users should be able to change the port. The best approach would be to use environment variables.
The `auth_token` needs to be secure. A default of `12345` is unacceptable if it's used for authentication.

Reply:
Can we listen on a unix socket perhaps? That way we can use file permissions to limit access.

Reply:
Whatever we do, let me know and I will create a change in SELinux policy. I prefer local socket to be honest, there is no need to expose this over TCP. I'd ditch token in this case since only puma/foreman/root users should be able to read it:

    activate_control_app 'unix:///var/run/foreman-pumactl.sock'

Oh it actually exposes also actions like start/stop/restart. We might consider tokens as well, could be stored in `/etc/foreman/settings.yml` and read by the CLI app. But that's security by obscurity rather than something good.

Reply:
Yes the sockets are better.  Now I added
`activate_control_app 'unix:///var/run/foreman/foreman-pumactl.sock'`
and puma automatically created the auth token for me.
```console
# cat /var/run/foreman/pids/puma.state 
---
pid: 7200
control_url: unix:///var/run/foreman/foreman-pumactl.sock
control_auth_token: 70443a6f423955719497c4c91df4ff
```